src/main/kotlin/com/avail/descriptor/representation/Descriptor.kt

```diff
@@ -0,0 +1,2733 @@
+		aContinuationType: A_Type) = false
+
+	override fun o_EqualsCompiledCodeType (
+		self: AvailObject,
+		aCompiledCodeType: A_Type) = false
+
+	override fun o_EqualsDouble (self: AvailObject, aDouble: Double) = false
+
+	override fun o_EqualsFloat (self: AvailObject, aFloat: Float) = false
+
+	override fun o_EqualsInfinity (self: AvailObject, sign: Sign) = false
+
+	override fun o_EqualsInteger (
+		self: AvailObject,
+		anAvailInteger: AvailObject) = false
+
+	override fun o_EqualsIntegerRangeType (
+		self: AvailObject,
+		another: A_Type) = false
+
+	override fun o_EqualsMap (self: AvailObject, aMap: A_Map) = false
+
+	override fun o_EqualsMapType (self: AvailObject, aMapType: A_Type) = false
+
+	override fun o_EqualsNybbleTuple (
+		self: AvailObject,
+		aTuple: A_Tuple) = false
+
+	override fun o_EqualsObject (
+		self: AvailObject,
+		anObject: AvailObject) = false
+
+	override fun o_EqualsObjectTuple (
+		self: AvailObject,
+		aTuple: A_Tuple) = false
+
+	override fun o_EqualsPhraseType (
+		self: AvailObject,
+		aPhraseType: A_Type) = false
+
+	override fun o_EqualsPojo (self: AvailObject, aPojo: AvailObject) = false
+
+	override fun o_EqualsPojoType (
+		self: AvailObject,
+		aPojoType: AvailObject) = false
+
+	override fun o_EqualsPrimitiveType (
+		self: AvailObject,
+		aPrimitiveType: A_Type) = false
+
+	override fun o_EqualsRawPojoFor (
+		self: AvailObject,
+		otherRawPojo: AvailObject,
+		otherJavaObject: Any?) = false
+
+	override fun o_EqualsReverseTuple (
+		self: AvailObject,
+		aTuple: A_Tuple) = false
+
+	override fun o_EqualsSet (self: AvailObject, aSet: A_Set) = false
+
+	override fun o_EqualsSetType (self: AvailObject, aSetType: A_Type) = false
+
+	override fun o_EqualsTupleType (
+		self: AvailObject,
+		aTupleType: A_Type) = false
+
+	override fun o_EqualsTwoByteString (
+		self: AvailObject,
+		aString: A_String) = false
+
+	override fun o_HasObjectInstance (
+		self: AvailObject,
+		potentialInstance: AvailObject): Boolean = unsupportedOperation()
+
+	override fun o_IsBetterRepresentationThan (
+		self: AvailObject,
+		anotherObject: A_BasicObject): Boolean
+	{
+		// Given two objects that are known to be equal, is the first one in a
+		// better form (more compact, more efficient, older generation) than the
+		// second one
+		val objectCost = self.objectSlotsCount() + self.integerSlotsCount()
+		val anotherCost =
+			anotherObject.objectSlotsCount() + anotherObject.integerSlotsCount()
+		return objectCost < anotherCost
+	}
+
+	override fun o_RepresentationCostOfTupleType (self: AvailObject): Int =
+		// Given two objects that are known to be equal, the second of which is
+		// in the form of a tuple type, is the first one in a better form than
+		// the second one?
+
+		// Explanation: This must be called with a tuple type as the second
+		// argument, but the two arguments must also be equal. All alternative
+		// implementations of tuple types should re-implement this method.
+		unsupportedOperation()
```

Comment:
Either leave this in braces, or move it into a Kdoc comment.  The code itself doesn't stand out well from the comment, and the paragraph gap makes it twice as hard to read.

Reply:
I made it a KDoc comment.